src/compiler/types.ts

```diff
@@ -3371,8 +3371,8 @@ namespace ts {
     export interface Symbol {
         flags: SymbolFlags;                     // Symbol flags
         escapedName: __String;                  // Name of symbol
-        declarations?: Declaration[];           // Declarations associated with this symbol
-        valueDeclaration?: Declaration;         // First value declaration of the symbol
+        declarations: Declaration[];           // Declarations associated with this symbol
+        valueDeclaration: Declaration;         // First value declaration of the symbol

```

Comment:
IIRC `valueDeclaration` is undefined for interfaces (and types?), is that still the case?

Reply:
Yes.

Reply:
Yes, the change was to remove a large number of assertions even if it meant having a less accurate type declaration.